File: test/test_zod2ts/class-expected/user.api.ts
// This is
// a multiline
// header.

import * as USER_DTOS from "./user.dtos";

export class ReqUpdateUser extends USER_DTOS.UpdateUserUseCaseDto {}

export class ResUpdateUser extends USER_DTOS.UpdateUserUseCaseResultDto {}

export class ResUpdateUserMulti {
    amount: number;
    data: USER_DTOS.UpdateUserUseCaseResultDto[];

    constructor(data: ResUpdateUserMulti) {
        this.amount = data.amount;
        this.data = data.data;
    }
}

export class UserApi {
    reqUpdateUser: ReqUpdateUser;
    resUpdateUser: ResUpdateUser;
    resUpdateUserMulti: ResUpdateUserMulti;

    constructor(data: UserApi) {
        this.reqUpdateUser = data.reqUpdateUser;
        this.resUpdateUser = data.resUpdateUser;
        this.resUpdateUserMulti = data.resUpdateUserMulti;
    }
}

File: test/test_zod2go/struct-expected/user.api.go
// This is
// a multiline
// header.

package models

import USER_DTOS "./user.dtos"

type ReqUpdateUser struct {
    USER_DTOS.UpdateUserUseCaseDto
}

type ResUpdateUser struct {
    USER_DTOS.UpdateUserUseCaseResultDto
}

type ResUpdateUserMulti struct {
    Amount int64 `json:"amount"`
    Data []USER_DTOS.UpdateUserUseCaseResultDto `json:"data"`
}

type UserApi struct {
    ReqUpdateUser ReqUpdateUser `json:"reqUpdateUser"`
    ResUpdateUser ResUpdateUser `json:"resUpdateUser"`
    ResUpdateUserMulti ResUpdateUserMulti `json:"resUpdateUserMulti"`
}
